core/src/consensus/mod.rs

```diff
@@ -106,9 +97,11 @@ pub trait ConsensusEngine<M: Machine>: Sync + Send {
     /// It is fine to require access to state or a full client for this function, since
     /// light clients do not generate seals.
     fn generate_seal(&self, _block: &M::LiveBlock, _parent: &M::Header) -> Seal {
```

Comment:
Is there a use case of this default implementation?

Reply:
Since PoW consensus uses an extra seal, it uses the default implementation of generate_seal.